src/lib.rs

```diff
@@ -647,6 +647,42 @@ impl Uuid {
         }
     }
 
+    /// Returns the four field values of the UUID.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use uuid::Uuid;
+    ///
+    /// let uuid = Uuid::nil();
+    /// assert_eq!(uuid.as_fields(), (0, 0, 0, [0; 8]));
+    ///
+    /// let uuid = Uuid::parse_str("936DA01F-9ABD-4D9D-80C7-02AF85C822A8").unwrap();
+    /// assert_eq!(uuid.as_fields(),
+    ///            (0x936DA01F, 0x9ABD, 0x4D9D, *b"\x80\xC7\x02\xAF\x85\xC8\x22\xA8"));
+    /// ```
+    pub fn as_fields(&self) -> (u32, u16, u16, [u8; 8]) {
+        let d1 = ((self.bytes[0] as u32) << 24) |
+                 ((self.bytes[1] as u32) << 16) |
+                 ((self.bytes[2] as u32) << 8) |
+                 (self.bytes[3] as u32);
+        let d2 = ((self.bytes[4] as u16) << 8) |
+                 (self.bytes[5] as u16);
+        let d3 = ((self.bytes[6] as u16) << 8) |
+                 (self.bytes[7] as u16);
+        let d4 = [
```

Comment:
Is the motivation behind `[u8; 8]` here that it's more technically correct and smaller in size? Personally I think `&[u8]` is a better choice though given the `&self` signature and symmetry with `from_fields`. It would also simplify the implementation here too.

What do you think?

Reply:
Mostly I was trying to match as closely as possible the signature of `as_bytes()`, which returns `&[u8; 16]`.  I would have liked to return `&[u8; 8]`, but I couldn't find a way to make that work, so I went with `[u8; 8]`.  But I'd be just as happy to go with `&[u8]`, if you prefer (and indeed it makes the implementation a bit simpler).

(The only other possible argument that occurs to me for returning `[u8; 8]` is that it's easier for the caller to turn that into `&[u8]` than it would be to turn a return value of `&[u8]` into a `[u8; 8]`, if the caller needs a static guarantee on the length for some reason.  But not sure how likely this is to matter -- for the use case I had in mind when I thought to add this method, a `&[u8]` works just fine.)

I'll update the PR.